### DISCUSSION TITLE
METplus Tutorial EXERCISE 1.2: Add_RH - Add Another Field To Grid_stat

### DISCUSSION THREAD

**Discussion by AustinConaty** [dtcenter/METplus#1346] [Configuration] [answered]
First time posting, so apologies in advance...   :)

I'm running the example on an AWS instance provided by DTC.

I'm looking at the exercise to add RH to an existing group of gridstat variables.  I noticed a typo on this web page:
https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-1-metplus-setupgrid-grid/end-practical-session-1-additional-exercises

In a Hint... the web page refers to GRID_STAT_OUTPUT_FLAG_SL1L2 = STAT   but in the config file in question, the
 GRID_STAT_OUTPUT_FLAG_SL1L2 should really be GRID_STAT_OUTPUT_FLAG_SAL1L2.  Either that or the config
file has a typo in it but I suspect it's the web page.  

<img width="886" alt="Screen Shot 2022-01-09 at 8 49 24 AM" src="https://user-images.githubusercontent.com/78030604/148685032-7284e9d8-8ca2-4afa-8609-a9bb6fb7e1a0.png">

The config file I'm referring to is
${METPLUS_TUTORIAL_DIR}/user_config/GridStat_add_rh.conf

So is the web page correct with GRID_STAT_OUTPUT_FLAG_SL1L2 or should it be GRID_STAT_OUTPUT_FLAG_SAL1L2 ?

pasting in some debug from a log file...
DEBUG 2: Processing RH/P500 versus RH/P500, for smoothing method NEAREST(1), over region FULL, using 10512 matched pairs.
DEBUG 2: Processing RH/P500 versus RH/P500, for smoothing method NEAREST(1), over region NHX, using 3600 matched pairs.
DEBUG 2: Processing RH/P500 versus RH/P500, for smoothing method NEAREST(1), over region SHX, using 3600 matched pairs.
DEBUG 2: Processing RH/P500 versus RH/P500, for smoothing method NEAREST(1), over region TRO, using 2448 matched pairs.
DEBUG 2: Processing RH/P500 versus RH/P500, for smoothing method NEAREST(1), over region PNA, using 1311 matched pairs.
DEBUG 2: Processing RH/P250 versus RH/P250, for smoothing method NEAREST(1), over region FULL, using 10512 matched pairs.
DEBUG 2: Processing RH/P250 versus RH/P250, for smoothing method NEAREST(1), over region NHX, using 3600 matched pairs.
DEBUG 2: Processing RH/P250 versus RH/P250, for smoothing method NEAREST(1), over region SHX, using 3600 matched pairs.
DEBUG 2: Processing RH/P250 versus RH/P250, for smoothing method NEAREST(1), over region TRO, using 2448 matched pairs.
DEBUG 2: Processing RH/P250 versus RH/P250, for smoothing method NEAREST(1), over region PNA, using 1311 matched pairs.

It looks like it's working, but when I go to look at the actual .stat files here:
 ls ${METPLUS_TUTORIAL_DIR}/output/exercises/add_rh/met_out/GFS/anom/*/*/*stat
The RH is not included in these files.

So when adding RH, it seems to process the data, but fails to write the expected output.

Many thanks,
METplus enthusiast
Austin Conaty

**Comment by CPKalb** [comment, marked as answer]
Hi Austin:

The hint is actually saying to add another line type, specifying the SL1L2 line in addition to the SAL1L2 line.  The SAL1L2 line requires a climatology to create output.  The RH data does not have a climatology so no RH output SAL1L2 lines can be produced.  To get RH output, it is necessary to add an additional output line.  The webpage example specifies SL1L2, but it could also be another line type (such as CNT, CTS, CTC, etc.).

Did that make sense?  I added some wording to the hint to try and make this more clear.

Thanks,
Christina

**Reply by AustinConaty** [reply]
I see.  Yes that worked for me and RH is now showing up in the *.stat files.
Thank you!
